csharp/src/Google.Protobuf/Collections/RepeatedField.cs

```diff
@@ -34,8 +34,8 @@ public sealed class RepeatedField<T> : IList<T>, IList, IDeepCloneable<RepeatedF
         private static readonly T[] EmptyArray = new T[0];
         private const int MinArraySize = 8;
 
-        private T[] array = EmptyArray;
-        private int count = 0;
+        internal T[] array = EmptyArray;
```

Comment:
Please keep all fields private. I have an alternative suggestion that doesn't require exposing any existing members.

I can see that we need a new type in order to "house" the extension method - but that can just call a new internal instance method on RepeatedField. That's more in the spirit of "really, we'd like it to be a normal instance method, but we don't want overload resolution issues." Keep all the actual *implementation* in RepeatedField, with an internal method of `AddSpan` or similar (it could even be a public method, potentially) and then just make the extension method delegate to that.

Reply:
Yeah, I agree. An internal `RepeatedField.AddRange(ReadOnlySpan<T>)` (or a different name if AddRange causes problems) extension method calls keeps things encapsulated.

Reply:
Addressed.